versioning/BRM/extentmap.cpp

```diff
@@ -1060,160 +1062,127 @@ void ExtentMap::mergeExtentsMaxMin(CPMaxMinMergeMap_t& cpMap, bool useLock)
-                   " lo_val=" << fExtentMap[i].partition.cprange.lo_val <<
-                   " min=" << it->second.min << " max=" << it->second.max <<
-                   " seq=" << it->second.seqNum;
+                os << "ExtentMap::mergeExtentsMaxMin(): casual partitioning update: firstLBID="
+                   << fExtentMap[i].range.start
+                   << " lastLBID=" << fExtentMap[i].range.start + fExtentMap[i].range.size * 1024 - 1
+                   << " OID=" << fExtentMap[i].fileID << " hi_val=" << fExtentMap[i].partition.cprange.hi_val
+                   << " lo_val=" << fExtentMap[i].partition.cprange.lo_val << " min=" << it->second.min
+                   << " max=" << it->second.max << " seq=" << it->second.seqNum;
                 log(os.str(), logging::LOG_TYPE_DEBUG);
 #endif
 
-                switch (fExtentMap[i].partition.cprange.isValid)
+                switch (emEntry.partition.cprange.isValid)
                 {
-                    // Merge input min/max with current min/max
-                    case CP_VALID:
+                // Merge input min/max with current min/max
+                case CP_VALID: {
+                    if (!isValidCPRange(it->second.max, it->second.min, it->second.type))
                     {
-                        if (!isValidCPRange( it->second.max,
-                                             it->second.min,
-                                             it->second.type ))
-                        {
-                            break;
-                        }
+                        break;
+                    }
 
-                        makeUndoRecord(&fExtentMap[i], sizeof(struct EMEntry));
+                    makeUndoRecordRBTree(UndoRecordType::DEFAULT, emEntry);
 
-                        // We check the validity of the current min/max,
-                        // because isValid could be CP_VALID for an extent
-                        // having all nullptr values, in which case the current
-                        // min/max needs to be set instead of merged.
+                    // We check the validity of the current min/max,
+                    // because isValid could be CP_VALID for an extent
+                    // having all NULL values, in which case the current
+                    // min/max needs to be set instead of merged.
 
-                        if (isValidCPRange(
-                                    fExtentMap[i].partition.cprange.hi_val,
-                                    fExtentMap[i].partition.cprange.lo_val,
-                                    it->second.type))
+                    if (isValidCPRange(emEntry.partition.cprange.hi_val, emEntry.partition.cprange.lo_val,
+                                       it->second.type))
+                    {
+                        // Swap byte order to do binary string comparison
+                        if (isCharType(it->second.type))
```

Comment:
This is an interesting part. Where did this come from? Do you recall the case that causes you to add this if branch?